packages/nimble-components/src/select/specs/filerable-select-hld.md

```diff
@@ -0,0 +1,97 @@
+-   Pressing `<Esc>` will close the dropdown and revert the value to what it was prior to opening dropdown.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [#1273: Filterable NimbleSelect](https://github.com/ni/nimble/issues/1273)
+-   [Filterable Select IxD](./IxD.md#filter)
+
+## Implementation / Design
+
+### API
+
+#### Select
+
+We will provide a means for clients to enable this feature with a new attribute:
+
+```ts
+export class Select() {
+    ...
+    @attr({ attribute: 'filter-mode' })
+    public filterMode = FilterMode.none;
+    ...
+}
+
+// Provide enum for filterMode to allow for future modes
+export const FilterMode  = {
+    none: 'none';
+    caseInsensitive: 'caseInsensitive';
+} as const;
+```
+
+#### LabelProviderCore
+
+We will be showing text that will require localization as part of this feature. It seems reasonable to add these APIs to the existing `LabelProviderCore` class as opposed to creating a new provider:
+
+```ts
+export class LabelProviderCore
+    ...
+    @attr({ attribute: 'select-filter-search' })
+    public selectFilterSearch: string | undefined;
+
+    @attr({ attribute: 'select-filter-no-results' })
+    public selectFilterNoResults: string | undefined;
+```
+
+The English strings used for these labels will be:
+
+-   selectFilterSearch: "Search"
+-   selectFilterNoResults: "No items found"
+
+### Implementation details
+
+As the filtering behavior is reminiscent of the `Combobox`, we will use the `fast-foundation` implementation for its `Combobox` as a guide for the changes needed for our `Select`. So, one interesting facet of this is that we must now provide an override for the `options` property, which now resembles [FAST's `Combobox` implementation](https://github.com/microsoft/fast/blob/8023f7ee8458ac147dee4dadb9b72ce45a142a1f/packages/web-components/fast-foundation/src/combobox/combobox.ts#L170):
+
+```ts
+public override get options(): ListboxOption[] {
+    Observable.track(this, 'options');
+    return this.filteredOptions?.length ? this.filteredOptions : this._options;
+}
+
+public override set options(value: ListboxOption[]) {
+    this._options = value;
+    Observable.notify(this, 'options');
+}
+```
+
+#### Filter Template
+
+Since the input for the filter is devoid of interaction semantics we find in the `NimbleTextField` (notably no green underline that appears on hover/focus), we will instead use a minimally styled native `input` element.
+
+#### Accessibility
+
+The accessibility tree will report that the search `input` element should have its own `focus` styling, however this should not be necessary, as the existence of the popup should provide the necessary focus hint to the user. Hiding the `input` from the accessibility tree using `aria-hidden` is strongly discouraged, and setting its `role` to `presentation` is not allowed.
+
+### Future considerations
```

Comment:
A couple other things to briefly think about and then probably just list here:
1. any need for debouncing the filter so it doesn't happen on each keystroke? I'm guessing not needed if there aren't too many items.
2. any considerations for list options being more dynamically populated?
   - do we need to fire any new events as filter text is entered? That might enable clients to do their own server-side filtering of items.
   - do we need to define behavior if the set of items changes while the popup is open?

Reply:
I've created a section in 'Future Considersations' related to dynamically fetched options. I felt there were enough unknowns coupled with the fact that I don't believe it interferes with or alters the current proposed API, that we don't need to design for this use-case at the moment.

Regarding debouncing, I don't believe there's anything that needs to be addressed specifically within this HLD.